jsonschema-plugin/src/test/groovy/io/micronaut/gradle/jsonschema/JsonSchemaGeneratorSpec.groovy

```diff
@@ -0,0 +1,213 @@
+package io.micronaut.gradle.jsonschema
+
+import io.micronaut.gradle.AbstractGradleBuildSpec
+import org.gradle.testkit.runner.TaskOutcome
+
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.StandardCopyOption
+
+class JsonSchemaGeneratorSpec extends AbstractGradleBuildSpec{
+    def "can generate source code from URL"() {
+        given:
+        allowSnapshots = true
+        settingsFile << "rootProject.name = 'jsonschema-url'"
+        buildFile << """
+            plugins {
+                id "io.micronaut.minimal.application"
+                id "io.micronaut.jsonschema"
+            }
+            
+            micronaut {
+                version "$micronautVersion"
+                jsonschema {
+                    url("https://www.hl7.org/fhir/fhir.schema.json") {
```

Comment:
Wondering if we should use a mock server to serve the schema, so that we don't depend on an external service for testing. We had occurrences of flaky build failures because of this in the past.